qt/scientific_interfaces/Direct/ALFAnalysisModel.cpp

```diff
@@ -88,28 +91,51 @@ CompositeFunction_sptr createCompositeFunction(IFunction_sptr const &flatBackgro
 namespace MantidQt::CustomInterfaces {
 
 ALFAnalysisModel::ALFAnalysisModel()
-    : m_function(createCompositeFunction(createFlatBackground(), createGaussian())), m_fitStatus(""), m_twoThetas() {}
+    : m_function(createCompositeFunction(createFlatBackground(), createGaussian())), m_fitStatus(""), m_twoThetas(),
+      m_extractedWorkspace() {}
+
+void ALFAnalysisModel::clear() {
+  m_extractedWorkspace = nullptr;
+  m_fitStatus = "";
+  m_twoThetas.clear();
+}
+
+void ALFAnalysisModel::setExtractedWorkspace(Mantid::API::MatrixWorkspace_sptr const &workspace,
+                                             std::vector<double> const &twoThetas) {
+  m_extractedWorkspace = workspace;
+  m_twoThetas = twoThetas;
+  m_fitStatus = "";
+}
+
+Mantid::API::MatrixWorkspace_sptr ALFAnalysisModel::extractedWorkspace() const { return m_extractedWorkspace; }
 
-void ALFAnalysisModel::doFit(std::string const &wsName, std::pair<double, double> const &range) {
+bool ALFAnalysisModel::isDataExtracted() const { return m_extractedWorkspace != nullptr; }
+
+MatrixWorkspace_sptr ALFAnalysisModel::doFit(std::pair<double, double> const &range) {
 
   IAlgorithm_sptr alg = AlgorithmManager::Instance().create("Fit");
   alg->initialize();
   alg->setProperty("Function", m_function);
-  alg->setProperty("InputWorkspace", wsName);
-  alg->setProperty("Output", wsName + "_fits");
+  alg->setProperty("InputWorkspace", m_extractedWorkspace);
+  alg->setProperty("Output", FIT_BASE_NAME);
   alg->setProperty("StartX", range.first);
   alg->setProperty("EndX", range.second);
   alg->execute();
+
   m_function = alg->getProperty("Function");
   m_fitStatus = alg->getPropertyValue("OutputStatus");
-}
+  auto const fitWorkspace = ADS.retrieveWS<MatrixWorkspace>(FIT_BASE_NAME + "_Workspace");
 
-void ALFAnalysisModel::calculateEstimate(std::string const &workspaceName, std::pair<double, double> const &range) {
-  auto &ads = AnalysisDataService::Instance();
-  if (ads.doesExist(workspaceName)) {
-    auto workspace = ads.retrieveWS<MatrixWorkspace>(workspaceName);
+  ADS.remove(FIT_BASE_NAME + "_Workspace");
```

Comment:
We seem to be adding/remove workspaces from the ADS a lot. Are they required to be in the ADS at all?

Reply:
I don't require them in ADS, and I've been trying to avoid using the ADS. I'm not sure if there is a way to tell the Fit algorithm to not output to the ADS? It seems different to other algorithms where you can do `StoreAlwaysInADS=False`, or maybe I am mistaken?

Reply:
All algorithms should respect that flag. Does `setChild(True)` make any difference?

If not then Fit must be adding the workspaces directly - would you have time to investigate?

Reply:
@martyngigg  Its all working now without the ADS, thanks!